servers/build-docker-images.sh

```diff
@@ -0,0 +1,215 @@
+  echo "    -r | --release-artifacts-dir <dir>    directory to place native binaries and uber-jars in" > /dev/stderr
+  echo "" > /dev/stderr
+  echo "GitHub mode is automatically enabled, when GITHUB_ENV is present. -i and -r are mandatory in GitHub mode." > /dev/stderr
+  echo "" > /dev/stderr
+  echo "Example: $0 nessie-unstable" > /dev/stderr
+}
+
+function gh_env() {
+  [ -n "${GITHUB_ENV}" ] && echo "$*" >> "${GITHUB_ENV}"
+  return 0
+}
+
+function gh_group() {
+  [ ${GITHUB} == 1 ] && echo "::group::$*" || (echo "" ; echo "** $*" ; echo "")
+}
+
+function gh_endgroup() {
+  [ ${GITHUB} == 1 ] && echo "::endgroup::" || echo ""
+}
+
+while [[ $# -gt 0 ]]; do
+  arg="$1"
+  case "$arg" in
+  -g | --github)
+    GITHUB=1
+    ;;
+  -i | --images-file)
+    IMAGES_FILE="$2"
+    shift
+    ;;
+  -r | --release-artifacts-dir)
+    RELEASE_FILES="$2"
+    shift
+    ;;
+  -h | --help)
+    usage
+    exit 0
+    ;;
+  -*)
+    usage
+    exit 1
+    ;;
+  *)
+    IMAGE_NAME="$arg"
+    ;;
+  esac
+  shift
+done
+
+if [[ ${GITHUB} == 1 ]] ; then
+  if [[ -z $IMAGES_FILE || -z $RELEASE_FILES ]] ; then
+    usage
+    exit 1
+  fi
+fi
+if [[ -z $IMAGE_NAME ]] ; then
+  usage
+  exit 1
+fi
+
+BASE_DIR="$(cd "$(dirname "$0")/.." ; pwd)"
+cd "$BASE_DIR"
+
+if [[ -z ${IMAGES_FILE} ]]; then
+  mkdir -p "$BASE_DIR/build/docker-images"
+  IMAGES_FILE="$(mktemp -p "$BASE_DIR/build/docker-images" images-XXXXX.txt)"
+  echo "Writing Docker images to push to:   $IMAGES_FILE"
+fi
+
+if [[ -z ${RELEASE_FILES} ]]; then
+  mkdir -p "$BASE_DIR/build/docker-images"
+  RELEASE_FILES="$(mktemp -p "$BASE_DIR/build/docker-images" -d release-files-XXXXX)"
+  echo "Placing binaries in:                $RELEASE_FILES"
+fi
+
+#
+# Prepare
+#
+
+gh_group "Prepare Docker image name and tag base"
+IMAGE_FULL_NAME="projectnessie/${IMAGE_NAME}"
+IMAGE_TAG="$(cat version.txt)"
+IMAGE_TAG_BASE="${IMAGE_TAG%-SNAPSHOT}"
+
+gh_env "IMAGE_NAME=${IMAGE_NAME}"
+gh_env "IMAGE_FULL_NAME=${IMAGE_FULL_NAME}"
+gh_env "IMAGE_TAG_BASE=${IMAGE_TAG%-SNAPSHOT}"
+
+echo "Image name: ${IMAGE_NAME}"
+echo "Full image name: ${IMAGE_FULL_NAME}"
+echo "Tag base: ${IMAGE_TAG_BASE}"
+
+[[ -n $IMAGES_FILE ]] && true > "${IMAGES_FILE}"
+mkdir -p "${RELEASE_FILES}"
+
+#
+# Native amd64 image
+#
+
+gh_group "Build native image"
+./gradlew :nessie-quarkus:clean :nessie-quarkus:quarkusBuild \
+  -Pnative \
+  -Pdocker \
+  -Dquarkus.jib.platforms=linux/amd64
+
+# Save the native runner binary in case we're publishing a release (not a snapshot)
+mv servers/quarkus-server/build/nessie-quarkus-*-runner "${RELEASE_FILES}"
+
+IMAGE_ID="$(cut -d: -f2 < servers/quarkus-server/build/jib-image.id)"
+echo "Native Linux amd64 image ID: ${IMAGE_ID}"
+gh_endgroup
+
+gh_group "Tagging ${IMAGE_ID} ..."
+for tag in "${IMAGE_TAG_BASE}" "${IMAGE_TAG_BASE}-native-amd64" "latest"; do
+  IMAGE_NAME_TAG="${IMAGE_FULL_NAME}:${tag}"
+  echo "Tagging ${IMAGE_ID} as ${IMAGE_NAME_TAG}"
+  docker tag "${IMAGE_ID}" "${IMAGE_NAME_TAG}"
```

Comment:
Would it make sense to also have a simple `VERSION` tag on `VERSION-native` for user convenience and continuity with old tagging scheme?